From: Mauro Carvalho Chehab <mchehab+huawei@kernel.org>
Date: Tue, 12 May 2020 07:51:56 +0200
Subject: ACPI / PMIC: Add i2c address for thermal control
Patch-mainline: v5.8-rc1
Git-commit: cc0594c4b0ef280dd7bdceab1a7c940d356f494d
References: jsc#SLE-16407

On Asus T101HA, we keep receiving those error messages:

	i915 0000:00:02.0: [drm] *ERROR* mipi_exec_pmic failed, error: -95
	intel_soc_pmic_exec_mipi_pmic_seq_element: Not implemented
	intel_soc_pmic_exec_mipi_pmic_seq_element: i2c-addr: 0x5e reg-addr 0x4b value 0x59 mask 0xff

Because the opregion is missing the I2C address.

Suggested-by: Hans de Goede <hdegoede@redhat.com>
Signed-off-by: Mauro Carvalho Chehab <mchehab+huawei@kernel.org>
Reviewed-by: Hans de Goede <hdegoede@redhat.com>
Reviewed-by: Mika Westerberg <mika.westerberg@linux.intel.com>
Reviewed-by: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Acked-by: Lee, Chun-Yi <jlee@suse.com>
---
 drivers/acpi/pmic/intel_pmic_chtdc_ti.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/acpi/pmic/intel_pmic_chtdc_ti.c
+++ b/drivers/acpi/pmic/intel_pmic_chtdc_ti.c
@@ -102,6 +102,7 @@ static struct intel_pmic_opregion_data c
 	.power_table_count = ARRAY_SIZE(chtdc_ti_power_table),
 	.thermal_table = chtdc_ti_thermal_table,
 	.thermal_table_count = ARRAY_SIZE(chtdc_ti_thermal_table),
+	.pmic_i2c_address = 0x5e,
 };
 
 static int chtdc_ti_pmic_opregion_probe(struct platform_device *pdev)
